spaceR/spaceR/ViewModels/RocketViewModel.swift

```diff
@@ -7,31 +7,53 @@
 
 import Foundation
 
+protocol ViewModelDelegate: AnyObject {
+    func reloadView()
+    func show(error: String)
+}
+
 class RocketViewModel {
     
-    private var allRockets: [Rocket]
+    private var repository: RocketRepositoryType?
+    private weak var delegate: ViewModelDelegate?
+    private var allRockets: [Rocket]?
     
-    init() {
+    init(repository: RocketRepositoryType,
+         delegate: ViewModelDelegate) {
+        self.repository = repository
+        self.delegate = delegate
         self.allRockets = []
     }
     
-    func setAllRockets(rockets: [Rocket]) {
+    func set(rockets: [Rocket]) {
```

Comment:
Same here, don't think you need this function you can set the variable directly 😄

Reply:
🤙🏼